org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/handler/ZigBeeThingHandler.java

```diff
@@ -363,10 +369,17 @@ private synchronized void doNodeInitialisation() {
                     continue;
                 }
 
-                logger.debug("{}: Initializing channel {} with {}", nodeIeeeAddress, channel.getUID(), handler);
-                if (handler.initializeDevice() == false) {
-                    logger.info("{}: Channel {} failed to initialise device", nodeIeeeAddress, channel.getUID());
-                    continue;
+                if (initializeDevice) {
+                    logger.debug("{}: Initializing channel {} with {}", nodeIeeeAddress, channel.getUID(), handler);
+                    if (handler.initializeDevice() == false) {
+                        logger.info("{}: Channel {} failed to initialise device", nodeIeeeAddress, channel.getUID());
+                        channelInitializationSuccessful = false;
+                        continue;
+                    }
+                } else {
+                    logger.debug(
+                            "{}: Initialization of channel {} with {} will be skipped as the device is already initialized",
```

Comment:
This log message sounds wrong, as the channel is still initialized (but only the converter part - just the device initialization part is skipped).

Correct would be something like "Device initialization for channel {} with {} is skipped as the device is already initialized".

IMHO, it would be even better to not log this for every channel, but rather use a single log message stating that device initialization is skipped for all channels (in the case that `initializeDevice` is false).

Reply:
I can remove this statement and put this one just before the loop:
```
            if (!initializeDevice) {
                logger.debug("{}: Device initialization will be skipped as the device is already initialized",
                        nodeIeeeAddress);
            }
```